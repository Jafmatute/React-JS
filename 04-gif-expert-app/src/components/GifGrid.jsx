import {getGifs} from '../helpers/getGifs.js';
import {useEffect} from "react";


const GifGrid = ({category}) => {

    useEffect(()=> {

        getGifs(category);

    },[])


    return (
        <>
            <h3>{category}</h3>
        </>
    );
};

export default GifGrid;
