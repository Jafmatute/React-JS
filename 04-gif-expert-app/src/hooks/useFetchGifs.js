import {useEffect, useState} from "react";
import {getGifs} from "../helpers/getGifs.js";

const UseFetchGifs = (category) => {

    const [images, setImages] = useState([]);
    const [isLoading, setIsLoading] = useState(true);

    useEffect(()=> {

        getGifs(category).then(img=> {
            setImages(img)
            setIsLoading(false);
        });

    },[])


    return {
        images,
        isLoading
    }
}

export default UseFetchGifs;