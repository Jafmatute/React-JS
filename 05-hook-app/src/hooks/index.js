export * from './useCounter.js';
export * from './useFecth.js';
export * from './useForm.js'