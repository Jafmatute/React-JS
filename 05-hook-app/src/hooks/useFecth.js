import {useEffect, useState} from "react";

const useFetch =(url) => {

    const [state, setState] = useState({
        data:null,
        isLoading:true,
        hasError:null
    });

    const getFech = async () => {

        setState({...state, isLoading: true})

        const rsp = await fetch(url);
        const data = await rsp.json();
        setState({
            data,
            isLoading: false,
            hasError: null,
        })

    }

    useEffect(() => {

        getFech();
    }, [url]);


    return {
        data: state.data,
        isLoading: state.isLoading,
        hasError: state.hasError
    }
}

export default useFetch;