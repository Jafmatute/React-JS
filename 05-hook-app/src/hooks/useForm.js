import {useState} from "react";

const UseForm = (initialForm={}) => {

    const [formState, setFormState] = useState(initialForm);

    const onInputChange = ({target}) => {

        const {name, value} = target;

        setFormState({
            ...formState,
            [name] : value
        })
    }

    return {
        ...formState,
    formState, onInputChange
    }
}
export default UseForm;