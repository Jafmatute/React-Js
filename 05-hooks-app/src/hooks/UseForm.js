import { useState } from 'react';

const UseForm = (initialState = {}) => {
  const [values, setValues] = useState(initialState);

  const handleInputChange = ({ target }) => {
    setValues({
      ...values,
      [target.name]: target.value,
    });
  };

  return [values, handleInputChange];
};

export default UseForm;
