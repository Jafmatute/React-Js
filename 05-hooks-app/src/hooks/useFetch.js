import { useEffect, useState } from 'react';

const useFetch = (url) => {
  const [state, setState] = useState({
    data: null,
    loading: null,
    error: null,
  });

  useEffect(() => {
    fetch(url)
      .then((rsp) => rsp.json())
      .then((data) => {
        setState({
          data,
          loading: false,
          error: null,
        });
      });
  }, [url]);

  return state;
};

export default useFetch;
