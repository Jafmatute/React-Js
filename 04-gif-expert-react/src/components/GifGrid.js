/** @format */

import React from 'react';
import PropTypes from 'prop-types';

import { useFetchGifs } from '../hooks/useFetchGifs';
import GifGridItem from './GifGridItem';

export const GifGrid = ({ categoria }) => {
  const { data: images, loading } = useFetchGifs(categoria);

  return (
    <>
      <h3 className='animate__animated animate__zoomIn'> {categoria} </h3>

      {loading && <p className='animate__animated animate__flash'>Loading</p>}

      <div className='card-grid'>
        {images.map((img) => (
          <GifGridItem key={img.id} {...img} />
        ))}
      </div>
    </>
  );
};
GifGrid.propTypes = {
  categoria: PropTypes.string.isRequired,
};

export default GifGrid;
