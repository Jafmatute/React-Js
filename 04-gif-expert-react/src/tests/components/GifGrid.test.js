import { shallow } from 'enzyme';
import GifGrid from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en el componente <GifGrid />', () => {
  const categoria = 'one Punch';

  test('Probar que renderice el componente correctamente', () => {
    useFetchGifs.mockReturnValue({ data: [], loading: true });
    const wrapper = shallow(<GifGrid categoria={categoria} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe de mostrar items cuando se cargan imagÃ©nes useFetchGifs', () => {
    const gifs = [
      {
        id: 'abc',
        url: 'https:/google.com',
        title: 'test',
      },
    ];
    useFetchGifs.mockReturnValue({ data: gifs, loading: false });
    const wrapper = shallow(<GifGrid categoria={categoria} />);

    expect(wrapper).toMatchSnapshot();
  });
});
