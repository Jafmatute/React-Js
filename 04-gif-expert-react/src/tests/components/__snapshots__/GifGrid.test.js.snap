// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GifGrid /> Debe de mostrar items cuando se cargan imag√©nes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__zoomIn"
  >
     
    one Punch
     
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="abc"
      key="abc"
      title="test"
      url="https:/google.com"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en el componente <GifGrid /> Probar que renderice el componente correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__zoomIn"
  >
     
    one Punch
     
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
